python/cudf/cudf/core/column/column.py

```diff
@@ -854,12 +854,16 @@ def as_categorical_column(self, dtype, **kwargs):
                 ordered=ordered,
             ).astype(dtype)
 
-        labels, cats = sr.factorize()
+        cats = sr.unique().astype(sr.dtype)
+        label_dtype = min_unsigned_type(len(cats))
+        labels = sr.label_encoding(
+            cats=cats, dtype=label_dtype, na_sentinel=np.iinfo(label_dtype).max
+        )
```

Comment:
using max as a sentinel value here makes me super uneasy, does this function not handle nulls natively?

Reply:
However this will not matter as the value will be masked later by the mask.